Toggl.Core/Extensions/ObservableExtensions.cs

```diff
@@ -49,5 +51,61 @@ public static IObservable<ITransition> OnErrorReturnResult<T>(this IObservable<I
             var signal = otherObservable.StartWith(Unit.Default);
             return observable.CombineLatest(signal, (t1, _) => t1);
         }
+
+        public static IObservable<T> TrackExecution<T>(this IObservable<T> observable, ActivityObserver activityObserver)
+        {
+            return activityObserver.trackActivityOfObservable(observable);
+        }
+    }
+
+    internal class ActivityToken<E> : IDisposable
+    {
+        private IObservable<E> _source;
+        private IDisposable _dispose;
+
+        internal ActivityToken(IObservable<E> source, BehaviorSubject<bool> running)
+        {
+            _source = source;
+            _dispose = Disposable.Create(() => running.OnNext(false));
+        }
+
+        internal IObservable<E> AsObservable()
+        {
+            return _source;
+        }
+
+        public void Dispose()
+        {
+            _dispose.Dispose();
+        }
+    }
+
+    public class ActivityObserver : IDisposable
```

Comment:
```suggestion
    public class ExecutionObserver : IDisposable
```

Activity is a very specific Android keyword and I'd rather avoid confusion